crates/wasmparser/src/validator.rs

```diff
@@ -1792,7 +1791,7 @@ mod arc {
                 return None;
             }
             debug_assert!(Arc::get_mut(&mut self.arc).is_some());
-            Some(unsafe { &mut *(&*self.arc as *const T as *mut T) })
+            Some(self.assert_mut())
```

Comment:
The downside of this change is that it introduces a runtime check.
The ideal replacement was [`Arc::get_mut_unchecked`](https://doc.rust-lang.org/alloc/sync/struct.Arc.html#method.get_mut_unchecked) but that's still unstable as of now.

Reply:
This needs to stay as it was, otherwise this is a mutally recursive function that never returns.

Reply:
Applied your suggestion and tests are running now again, too. :tada: